core/src/main/scala/org/apache/spark/scheduler/ExecutorFailuresInTaskSet.scala

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.scheduler
+
+import scala.collection.mutable.HashMap
+
+/**
+ * Small helper for tracking failed tasks for blacklisting purposes.  Info on all failures for one
+ * task set, within one task set.
+ */
+class ExecutorFailuresInTaskSet(val node: String) {
```

Comment:
I'd like to push back on this and keep this naming.  I think I actually came up with this name from an earlier comment of yours about being confused about some variables and asking for more particular names :) (maybe it was for other variables).

the naming could just be on instance variables and not on the class itself of course, but I think its helpful in this case to have it on the class.  I found even i got easily confused about the purpose of various little helper classes when I had to come back to this patch after a couple of days.  Where its used generally told me more about the important design tradeoffs -- eg., if its in a taskset, it'll get dropped when the taskset completes, which automatically takes care of memory leaks.  Maybe at some point in the future it'll be used for something else, but it seems like then that change go do the appropriate renaming.  I feel like the name would also make it clear to the later change to check existing assumptions about the scope of this helper.

It doesn't work as a nested class in the larger blacklist patch as I currently have it, so since you don't feel strongly I'll just leave at the top level (but I will make it `private[scheduler]`, which it should have been already).